### ISSUE TITLE
unable to mount volume for the pods from velero

### ISSUE THREAD

**Issue by prasadcz** [vmware-tanzu/velero#7513]
pod error im getting - 
![image](https://github.com/vmware-tanzu/velero/assets/116771789/f4715aca-82fd-436f-abf8-ec6438b275d1)
  Warning  FailedAttachVolume  36s  attachdetach-controller  AttachVolume.Attach failed for volume "pvc-86238689-c2d2-4c10-bf74-b3d23f53d56b" : rpc error: code = Internal d
esc = Could not get volume with ID "vol-01cf0a57814261ddb": InvalidVolume.NotFound: The volume 'vol-01cf0a57814261ddb' does not exist.
           status code: 400, request id: 6662a538-3713-44f4-817a-7769eb43010b
  Warning  FailedAttachVolume  3s  attachdetach-controller  AttachVolume.Attach failed for volume "pvc-86238689-c2d2-4c10-bf74-b3d23f53d56b" : rpc error: code = Internal de
sc = Could not get volume with ID "vol-01cf0a57814261ddb": InvalidVolume.NotFound: The volume 'vol-01cf0a57814261ddb' does not exist.
           status code: 400, request id: 19f0aaf5-214c-42a3-b96c-002aa7350058
- i have checked pv and pvc pv is created and it is bound to pvc the issue here the volume id which im getting is pv is incorrect and i cant find that volume id in aws console, but my velero is able to take backup and create snapshots

Use case im trying is - taking the backup of namespace xyz and deleting all the resources nd namespace xyz and again im restoring it using velero in same eks cluster

**Comment by danfengliu** [comment]
Hi @prasadcz , could you provide Velero debug bundle using Velero CLI as below? If you can not provide that due to some sensitive data reason, could you provide more details of the backup? 

`velero debug --namespace  <your-velero-ns> --output debug-bundle-xxx.tar.gz --verbose`

`velero describe backups  <your-backup>  -n <your-velero-ns>  --details `

**Comment by prasadcz** [comment]
@danfengliu  debug bundle if you need - 
[debug-bundle.tar.gz](https://github.com/vmware-tanzu/velero/files/14570573/debug-bundle.tar.gz)

**Comment by danfengliu** [comment]
@prasadcz I can not download this bundle file, it reports 404, could you check that? From the backup, I see you had a backup using AWS native snapshot, we do need Velero log to see what's wrong when restoring this backup.

**Comment by prasadcz** [comment]
@danfengliu 
[debug-bundle-xxx.tar.gz](h
[debug-bundle-xxx.tar.gz](https://github.com/vmware-tanzu/velero/files/14583442/debug-bundle-xxx.tar.gz)
[restore.log](https://github.com/vmware-tanzu/velero/files/14583481/restore.log)
[backup.log](https://github.com/vmware-tanzu/velero/files/14583490/backup.log)
ttps://github.com/vmware-tanzu/velero/files/14583426/debug-bundle-xxx.tar.gz)

**Comment by prasadcz** [comment]
[restore.log](https://github.com/vmware-tanzu/velero/files/14583563/restore.log)

**Comment by prasadcz** [comment]
@danfengliu  i have uploaded the files of restore logs backup logs and debug logs in my public repo as im unable to upload it here so please clone or download files from that repo - 
https://github.com/prasadcz/Test/blob/master/backup.log
https://github.com/prasadcz/Test/blob/master/restore.log
https://github.com/prasadcz/Test/blob/master/debug-bundle-xxx.tar.gz

**Comment by danfengliu** [comment]
What's your storage class look like? Could you provide that?

**Comment by blackpiglet** [comment]
```time="2024-03-11T11:28:39Z" level=warning msg="No volume ID returned by volume snapshotter for persistent volume" backup=velero/darwin-backup logSource="pkg/backup/item_backupper.go:600" name=jenkins-pv namespace= persistentVolume=jenkins-pv resource=persistentvolumes volumeSnapshotLocation=default```

Please check the backed-up PV's content. By far, the Velero AWS plugin only support the CSI and the EBS volume.

**Comment by prasadcz** [comment]
@danfengliu  @blackpiglet  
![image](https://github.com/vmware-tanzu/velero/assets/116771789/9dc5d17e-c7f4-4913-b823-4d188c9dd6c6)
this is my storage class describe and also for pv retain policies i have edited manually to delete while taking backup and also for @blackpiglet question snapshot status is available but the thing is volume id which is fetching while restoring is invalid

**Comment by blackpiglet** [comment]
@prasadcz 

Could you describe the PV?

**Comment by blackpiglet** [comment]
I think the error was introduced in the PV backup.
The snapshot was not created correctly. 
No error returned for snapshot creation was misleading.
Could you check the content of the backup-created snapshot in AWS? If no volume ID was provided, not sure what the snapshot looks like.
```
time="2024-03-11T11:28:39Z" level=warning msg="No volume ID returned by volume snapshotter for persistent volume" backup=velero/darwin-backup logSource="pkg/backup/item_backupper.go:600" name=jenkins-pv namespace= persistentVolume=jenkins-pv resource=persistentvolumes volumeSnapshotLocation=default
```

**Comment by danfengliu** [comment]
@prasadcz We infered from the Velero logs and your storage class that Velero is working well and there should be issue of snapshot you're taken, and we suggest you create volume with snapshot of your backup on AWS web console to see will  there be a successful volume creation.

![image](https://github.com/vmware-tanzu/velero/assets/44018302/ea22e42d-8b34-4242-a11c-4c80e398f851)

**Comment by danfengliu** [comment]
Is there any way to trace when and who delete the volume created by snapshot triggered by Velero restore?

**Comment by prasadcz** [comment]
@danfengliu this issue is sorted when  i gave ec2 full access for velero role thanks for the support

**Comment by danfengliu** [comment]
Indeed, in this case, Velero have no awareness of the volume status, if you can provide the problematic volume error info during your investigation on AWS web console, that would be helpful for us to make a decision of improving Velero user experience in these scenario.

**Comment by Skaronator** [comment]
I ran into the exact same issue. Created backup, deleted Pod and PVC / PV and then try to restore it resulting in the exact same error.

After lots of troubleshooting I found a solution. Its definitely a permission issue but the velero log wasn't helpful. Giving full admin access fixed this. I'm now trying to further scope it down.

It's something with missing KMS permissions since my volumes are encrypted.

**Comment by Skaronator** [comment]
I was able to scope it down to these permissions:

```hcl
data "aws_iam_policy_document" "velero" {
  statement {
    actions = [
      "kms:CreateGrant",
      "kms:ListGrants",
      "kms:RevokeGrant",
      "kms:RetireGrant",
      "kms:ListRetirableGrants",
      "kms:ReEncryptFrom",
      "kms:ReEncryptTo",
    ]
    resources = [aws_kms_key.ebs.arn]
  }
}
```

Probably some Grant permissions can be also removed but there may be an edge case where they are needed.
